db/models.py

```diff
@@ -50,3 +51,42 @@ class MovieSession(models.Model):
 
     def __str__(self) -> str:
         return f"{self.movie.title} {str(self.show_time)}"
+
+
+class Order(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True)
+    user = models.ForeignKey(
+        to="db.User", on_delete=models.CASCADE, related_name="orders"
+    )
+
+    def __str__(self) -> str:
+        return str(self.created_at)
+
+
+class Ticket(models.Model):
+    row = models.IntegerField()
+    seat = models.IntegerField()
+    movie_session = models.ForeignKey(
+        to=MovieSession, on_delete=models.CASCADE, related_name="tickets"
+    )
+    order = models.ForeignKey(
+        to=Order, on_delete=models.CASCADE, related_name="tickets"
+    )
+
+    def __str__(self) -> str:
+        return f"{self.movie_session.movie.title} {str(self.movie_session.show_time)} (row: {self.row}, seat: {self.seat})"
+
+    def clean(self):
+        if self.row > self.movie_session.cinema_hall.rows:
+            raise ValidationError({'row': ['row number must be in available range: (1, rows): (1, 18)']})
+        if self.seat > self.movie_session.cinema_hall.seats_in_row:
+            raise ValidationError({'seat': ['seat number must be in available range: (1, seats_in_row): (1, 24)']})
+
+        if self.row < 1 or self.row > self.movie_session.cinema_hall.rows:
+            raise ValidationError({'row': ['row number must be in available range: (1, rows): (1, 18)']})
+        if self.seat < 1 or self.seat > self.movie_session.cinema_hall.seats_in_row:
+            raise ValidationError({'seat': ['seat number must be in available range: (1, seats_in_row): (1, 24)']})
```

Comment:
The validation checks for `row` and `seat` in the `clean` method are redundant. You are checking the same conditions twice: once in lines 80-83 and again in lines 85-88. You can remove the first set of checks (lines 80-83) as they are covered by the second set (lines 85-88).